Controllers/PlayingController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Acuedify.Models;
+using Acuedify.Services.Library.Interfaces;
+using Acuedify.Services.Playing;
+using Acuedify.Services.Playing.Interfaces;
+using Azure.Messaging;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Http;
+
+namespace Acuedify.Controllers
+{
+    [Route("Play")]
+    public class PlayingController : Controller
+	{
+        private readonly ILibraryService _libraryService;
+		private readonly IPlayingService _playingService;
+
+		public PlayingController(ILibraryService libraryService, IPlayingService playingService)
+        {
+            _libraryService = libraryService;
+            _playingService = playingService;
+        }
+
+        // GET:
+        [HttpGet]
+        public IActionResult Index(int quizId, int questionId)
+        {
+		    PlayDetails details;
+
+            if (questionId == 0) // Initial loading of the screen with the first flashcard
+            {
+                var flashcardSet = _libraryService.GetUserQuiz(quizId);
+				var flashcards = _libraryService.GetQuizQuestions(quizId);
+
+                flashcards.Shuffle();
+
+				if (flashcards == null)
+				{
+					return View("ErrorView", "no quiz found by id");
+				}
```

Comment:
From validation it seems `flashcards` can be null. If so, won't calling `flashcards.Shuffle()` might throw NullReferenceException ?

Reply:
fixed